contrib/ros/util/common.py

```diff
@@ -3,29 +3,24 @@
 import time
 
 
-def rate_limited(max_per_second: int):
+def rate_limited(func):
     """Rate-limits the decorated function locally, for one process."""
     lock = threading.Lock()
-    min_interval = 1.0 / max_per_second
+    last_time_called = time.perf_counter()
 
-    def decorate(func):
-        last_time_called = time.perf_counter()
+    @wraps(func)
+    def rate_limited_function(*args, **kwargs):
+        lock.acquire()
+        nonlocal last_time_called
+        try:
+            elapsed = time.perf_counter() - last_time_called
+            left_to_wait = 1.0 / args[0].step_freq - elapsed
```

Comment:
this is an abuse of metaprogramming. please remove and use an argument, as before.